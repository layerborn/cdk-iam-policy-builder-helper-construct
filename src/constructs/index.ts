export * from './Actions';
export * from './ManagedPolicies';
