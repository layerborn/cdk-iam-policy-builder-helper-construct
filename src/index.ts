import * as actions from './construct/Actions.json';
import * as policies from './construct/ManagedPolicies.json';

export const Actions = actions;
export const ManagedPolicies = policies;
