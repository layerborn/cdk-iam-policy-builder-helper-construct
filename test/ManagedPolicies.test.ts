import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { ManagedPolicy, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
import { ManagedPolicies } from '../src';

describe('Smoke Test ManagedPolicies Helper Properties', () => {
  test('Smoke Test Managed Policies', () => {
    console.log(ManagedPolicies.AmazonElastiCacheFullAccess.PolicyName);
    expect(ManagedPolicies.AmazonElastiCacheFullAccess.PolicyName).toBe('AmazonElastiCacheFullAccess');
    expect(ManagedPolicies.AmazonElastiCacheFullAccess.Arn).toBe('arn:aws:iam::aws:policy/AmazonElastiCacheFullAccess');
    expect(ManagedPolicies.AmazonFSxConsoleFullAccess.PolicyName).toBe('AmazonFSxConsoleFullAccess');
    expect(ManagedPolicies.AmazonFSxConsoleFullAccess.Arn).toBe('arn:aws:iam::aws:policy/AmazonFSxConsoleFullAccess');
  });

  test('Create an IAM Role using Actions and Managed Policies', () => {
    const app = new App();
    const fakeEnv = {
      account: '123456789012',
      region: 'eu-central-1',
    };
    const stack = new Stack(app, 'MyStack', {
      stackName: 'MyStackManagedPolicies',
      env: fakeEnv,
    });
    new Role(stack, 'MyRole', {
      roleName: 'MyRole',
      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
      managedPolicies: [
        ManagedPolicy.fromManagedPolicyArn(stack, 'LambdaBasicExecutionRolePolicy', ManagedPolicies.AWSLambdaBasicExecutionRole.Arn),
        ManagedPolicy.fromManagedPolicyName(stack, 'LambdaVpcPolicy', ManagedPolicies.AWSLambdaVPCAccessExecutionRole.PolicyName),
      ],
    });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
