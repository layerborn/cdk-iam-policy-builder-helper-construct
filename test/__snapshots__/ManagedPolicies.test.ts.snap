// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Test ManagedPolicies Helper Properties Create an IAM Role using Actions and Managed Policies 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "MyRoleF48FFE04": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::123456789012:policy/AWSLambdaVPCAccessExecutionRole",
              ],
            ],
          },
        ],
        "RoleName": "MyRole",
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
